Assets/Scripts/StandardSamples/UI/Common/SampleSelectableStateHandler.cs

```diff
@@ -0,0 +1,108 @@
+/*
+* Copyright (c) 2021 PlayEveryWare
+* 
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+* 
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    using System.Collections;
+    using UnityEngine;
+    using UnityEngine.UI;
+    using UnityEngine.EventSystems;
+    using System;
+
+    public class SampleSelectableStateHandler : MonoBehaviour
+    {
+        public struct SampleInteractableNewState
+        {
+            public bool Interactable;
+            public string NewTooltipText;
+
+            public SampleInteractableNewState(bool shouldBeInteractable, string newTooltipText = "")
+            {
+                Interactable = shouldBeInteractable;
+                NewTooltipText = newTooltipText;
+            }
+        }
+
+        public delegate SampleInteractableNewState GetSampleInteractable();
+
+        private static event Action UpdateSelectableStates;
+
+        private GetSampleInteractable sampleInteractableFunction { get; set; }
```

Comment:
This is confusing to me for a number of reasons:
1. It's a private property, but has the word function in it
2. The return type has the word Get in it.

In the very least I think it needs some documentation around it for clarity.

Reply:
The naming is pretty confusing, eh? The intention here is supposed to be,

- When this Handler is told to update, run this function.
- The return function will answer if the object should be interactable. If it is, make the attached behavior interactable.
- Otherwise, set the tooltip to this value.

This function is provided by other classes that point to specific UI Elements, giving them specific implementations of this function. So each of these SampleSelectableStateHandlers needs a reference to their own function.

Thus `GetSampleInteractable` should be answering "is this sample element interactable?" What do you suppose we should rename it to?